File: 30-days-of-code/java/day-23-bst-level-order-traversal.java
import java.util.*;
import java.io.*;
class Node{
    Node left,right;
    int data;
    Node(int data){
        this.data=data;
        left=right=null;
    }
}
class Solution{

static void levelOrder(Node root){
        Queue<Node> q = new LinkedList<>();
        q.add(root);
        
        while (!q.isEmpty()) {
            Node curr_node = q.remove();
            System.out.print(String.format("%d%s", curr_node.data, " "));
            
            if (curr_node.left != null) {
                q.add(curr_node.left);
            }
            
            if (curr_node.right != null) {
                q.add(curr_node.right);
            }
        }
      
    }

public static Node insert(Node root,int data){
        if(root==null){
            return new Node(data);
        }
        else{
            Node cur;
            if(data<=root.data){
                cur=insert(root.left,data);
                root.left=cur;
            }
            else{
                cur=insert(root.right,data);
                root.right=cur;
            }
            return root;
        }
    }
    public static void main(String args[]){
            Scanner sc=new Scanner(System.in);
            int T=sc.nextInt();
            Node root=null;
            while(T-->0){
                int data=sc.nextInt();
                root=insert(root,data);
            }
            levelOrder(root);
        }	
}

File: 30-days-of-code/cpp/day-23-bst-level-order-traversal.cpp
#include <iostream>
#include <cstddef>
#include <queue>
#include <string>
#include <cstdlib>

using namespace std;	
class Node{
    public:
        int data;
        Node *left,*right;
        Node(int d){
            data=d;
            left=right=NULL;
        }
};
class Solution{
    public:
        Node* insert(Node* root, int data){
            if(root==NULL){
                return new Node(data);
            } else {
                Node* cur;
                if(data<=root->data){
                    cur=insert(root->left,data);
                    root->left=cur;
                } else {
                   cur=insert(root->right,data);
                   root->right=cur;
                }           
           return root;
           }
        }

	void levelOrder(Node * root){
        // Begin Solution
        //---------------------------------------------------------------------------------------
        queue<Node *> q;
        q.push(root);

        while (q.size() != 0) {
            Node *curr_node = q.front();
            q.pop();

            cout << curr_node->data << " ";

            if (curr_node->left) {
                q.push(curr_node->left);
            }
            if (curr_node->right) {
                q.push(curr_node->right);
            }
        }
        //---------------------------------------------------------------------------------------
        // End Solution
  	}

};

int main() {
    Solution myTree;
    Node* root=NULL;
    int T,data;
    cin>>T;
    while(T-->0){
        cin>>data;
        root= myTree.insert(root,data);
    }
    myTree.levelOrder(root);
    return 0;
}